code/oEmbeddableFields.php

```diff
@@ -7,387 +7,537 @@
+    ];
 
-    private static $db = array(
-        'SlideshareID'		        => 'Int',
-        'MixcloudURL'		        => 'VarChar(511)',
-        'YouTubeID'			        => 'VarChar(127)',
-        'BrainsharkID'		        => 'VarChar(127)',
-        'VidyardID'			        => 'VarChar(127)',
-    );
+    /**
+     * List of one-to-one relationships. {@link DataObject::$has_one}
+     *
+     * @var array
+     */
+    private static $has_one = [
+        'Image' => 'Image'
+    ];
 
-    private static $indexes = array();
+    private static $indexes = [];
 
 
-	private function TypeEnabled($type = false){
+    private function TypeEnabled($type = false)
+    {
 
-		if(!is_array($this->owner->config()->enabled_oembed_fields)) return true;
+        if (!is_array($this->owner->config()->enabled_oembed_fields)) return true;
 
-		if(in_array($type, $this->owner->config()->enabled_oembed_fields)){
-			return true;
-		}
+        if (in_array($type, $this->owner->config()->enabled_oembed_fields)) {
+            return true;
+        }
 
-		return false;
-	}
+        return false;
+    }
 
     /**
      * @param \FieldList $fields
      */
-    public function updateCMSFields(FieldList $fields) {
+    public function updateCMSFields(FieldList $fields)
+    {
 
-        $Media = array();
+        $Media = [];
+
+        if ($this->TypeEnabled('Image')) {
+            $Media['Image'] = UploadField::create('Image',
+                'Image');
+            $Media['Image']->setRightTitle(
+                'You must first save this Media before adding the image');
+        } else {
+            $Media['Image'] = HiddenField::create('Image', '');
+        }
 
-        if($this->TypeEnabled('Slideshare')){
-	        $Media['SlideshareID'] = TextField::create('SlideshareID',
-	            '<span style="position:relative;">
+        if ($this->TypeEnabled('Slideshare')) {
+            $Media['SlideshareID'] = TextField::create('SlideshareID',
+                '<span style="position:relative;">
 	                <a href="http://Slideshare.com/upload" target="_blank" style="position:absolute;top:1.5em;white-space:nowrap;">Upload Slides</a>
 	            </span>
-	            Slideshare URL',$this->owner->SlideshareID ?: null);
-	        $Media['SlideshareID']->setRightTitle(
-	            'Include the Slideshare <strong>URL</strong> for slideshows.<br/>'.
-	            '<strong>After saving:</strong> If valid, this will automatically convert to the Slideshare ID. If not, it will change to "0".'
-	        //TO GET IT: <u>Drag</u> this <a href="javascript:alert(typeof slideshare_object == \'object\' ? slideshare_object.presentationId : \'Not on a Slideshare page!\');">[SlideshareID]</a> to your bookmarks, then click the button while viewing a Slideshare page.'
-	        );
-        }else{
-	        $Media['SlideshareID'] = HiddenField::create('SlideshareID', '');
+	            Slideshare URL', $this->owner->SlideshareID ?: null);
```

Comment:
If a call is too long to fit in one line "readably", each parameter should have its own line.